### DISCUSSION TITLE
Support reading SAV files from memory

### DISCUSSION THREAD

**Discussion by janwillemvl** [Roche/pyreadstat#157] [Ideas]
As far as I can see, currently the read_sav function only supports reading in a file from local drive. My script does not have writing rights on local drive, and the idea is that it reads a sav file from S3, parses it and writes the parsed contents to a RDS database in AWS. This would work if the function read_sav would accept an in-memory-file (BytesIO object).

**Comment by ofajardo** [comment]
Yes, I agree it would be very useful. Unfortunately it is not possible at the moment as the underlying C library Readstat does not allow that.A PR has already submitted but has not progressed so far. You can read more details here: 
https://github.com/Roche/pyreadstat/issues/8
https://github.com/WizardMac/ReadStat/pull/179

**Reply by janwillemvl** [reply]
Thanks for the quick reply. From the looks of it, some integration tests failed and nobody has picked it up for over a year. So probably nobody will pick it up, unfortunately. I don't think I can do anything to help, right? I'll look for an alternative to reading from memory like using a temporary file storage mounted in the docker container we use.

**Comment by ofajardo** [comment]
Yes, it probably will take a long time before it is solved, if it gets solved at all. So you need to arrange some alternative for your work if you want to progress

**Reply by janwillemvl** [reply]
Ok, thanks for your support anyway!
